src/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeFileUpload.tsx

```diff
@@ -90,18 +185,54 @@ export const DefendDisputeFileUpload = () => {
-                        >
-                            {i18n.get(document.documentTypeCode as TranslationKey)}
-                        </FileInput>
-                    ))}
+                    <div className="adyen-pe-defend-dispute-document-upload-box">
+                        <div className="adyen-pe-defend-dispute-document-upload-box__required-documents">
+                            {requiredDocuments?.map(document => {
+                                return (
+                                    <DefendDocumentUpload
+                                        key={document}
+                                        document={document}
+                                        ref={ref}
+                                        addFileToDefendPayload={addFileToDefendPayload}
+                                        isRequired
+                                    />
+                                );
+                            })}
+                        </div>
+
+                        {oneOrMoreDocuments.length ? (
+                            <SelectAndUploadOptionalDoc
+                                selection={oneOrMoreSelectedDocument}
+                                setSelection={(val: string) => setOneOrMoreSelectedDocument(val)}
+                                items={oneOrMoreDocuments}
+                                ref={ref}
+                                title={i18n.get('disputes.extraRequiredDocuments')}
+                                required
+                            />
+                        ) : null}
+                    </div>
+                    {optionalSelectedDocuments.length
+                        ? optionalSelectedDocuments.map((doc, index) => {
+                              return (
+                                  <div key={`optional-doc-${index}`} className="adyen-pe-defend-dispute-document-upload-box">
+                                      <SelectAndUploadOptionalDoc
+                                          onRemoveOption={removeSelectedOptionalDocument}
+                                          selection={doc}
+                                          setSelection={addOptionalDocument}
+                                          index={index}
+                                          items={availableOptionalDocuments}
+                                          ref={ref}
+                                          title={i18n.get('disputes.optionalDocument')}
+                                          required={false}
+                                      />
+                                  </div>
+                              );
+                          })
+                        : null}
+                    <Button onClick={addEmptyOptionalDocument} variant={ButtonVariant.SECONDARY} fullWidth align="center">
+                        <Icon name="plus" />
+                        {i18n.get('disputes.addOptionalDocument')}
+                    </Button>
```

Comment:
Consider disabling the **Add optional document** button when no more optional document can be uploaded (the user has exhausted the available number of applicable documents). Here are some suggestions on how to achieve this:

1. Compute a value to determine the availability of optional documents.
```ts
const canAddOptionalDocument = useMemo(() => {
    const optionalDocumentsCount = optionalDocuments.length + Math.max(0, oneOrMoreDocuments.length - 1);
    return optionalDocumentsCount && optionalDocumentsCount !== optionalSelectedDocuments.length;
}, [oneOrMoreDocuments, optionalDocuments, optionalSelectedDocuments]);
```

2. Using the `canAddOptionalDocument` determine whether to disable the button.
```suggestion
                    <Button
                        align="center"
                        disabled={!canAddOptionalDocument}
                        onClick={addEmptyOptionalDocument}
                        variant={ButtonVariant.SECONDARY}
                        fullWidth
                    >
                        <Icon name="plus" />
                        {i18n.get('disputes.addOptionalDocument')}
                    </Button>
```

Reply:
✅